packages/design-system/src/LoginButton.tsx

```diff
@@ -0,0 +1,85 @@
+import {
+  Avatar,
+  Box,
+  Button,
+  HStack,
+  Link,
+  Spinner,
+  Text,
+} from '@chakra-ui/react';
+import React, { useCallback } from 'react';
+
+import { MetaButton } from './MetaButton';
+
+type UserInfo = {
+  playerImage?: string;
+  playerName?: string;
+  username?: string;
+  player?: any;
+};
+
+type LoginButtonPros = {
+  connectWeb3: () => Promise<void>;
+  disconnect: () => void;
+  isConnected: boolean;
+  user: UserInfo;
+  fetching: boolean;
+};
+
+export const LoginButton: React.FC<LoginButtonPros> = ({
+  connectWeb3,
+  disconnect,
+  isConnected,
+  user,
+  fetching,
+}: LoginButtonPros) => {
+  const handleLoginClick = useCallback(async () => {
+    await connectWeb3();
+  }, [connectWeb3]);
+
+  if (isConnected) {
+    if (fetching) {
+      return <Spinner color="purple.500" size="sm" />;
+    }
+    if (!user?.player) return null;
+    return (
+      <HStack>
+        <Avatar
+          src={user.playerImage}
+          name={user.playerName}
+          as={Link}
+          href={`/player/${user.username}`}
+        />
+        <Box>
+          <Link
+            href={`/player/${user.username}`}
+            fontFamily="body"
+            color="whiteAlpha.700"
+          >
+            {user.player ? user.playerName : 'Unknown'}
+          </Link>
+          <HStack spacing={2}>
+            <Link color="cyan.400" href="/profile/setup">
```

Comment:
This has implementation details that don't belong in the design system (profile setup route has to do with NextJS in the `web` package) since they cant be universally used. Would be better to just use the generic button component or abstract the implementation details of the NextJS app.

Reply:
The tricky part is whether to always include the `Disconnect` button. It is arguably always warranted, but, for example, I recently pushed a change to MyMeta to replace the `Disconnect` text with a `⨯`.

Maybe `mutations` could be a function with a default value of the `Disconnect` button, but which receives the `disconnect` function as a parameter allowing the generation of whatever UI is desired.

Is that too convoluted?